packages/ts/src/components/donut/types.ts

```diff
@@ -1,10 +1,14 @@
 import { PieArcDatum } from 'd3-shape'
 
+export type DonutInputArc<Datum> = Partial<DonutArcDatum<Datum>> & {
```

Comment:
@reb-dev I'm a bit concerned about this type. From the code (see below) it looks like a wrapper around `Datum`
 and it doesn't have information about arcs (i.e. `startAngle`, `endAngle`, ...):
 
```ts
    const data: DonutInputArc<Datum>[] = datamodel.data.reduce((acc, d, i) =>
      !config.showEmptySegments && !getNumber(d, config.value, i)
        ? acc : [...acc, { _datum: d, _index: i }]
    , [])
```

So I would simply call it something like `DonutDatum` and define as:
```ts
export type DonutDatum<Datum> = Datum & {
  _index: number;
}
```
One can argue that original `Datum` can already have a property called `_index` and that will be a valid argument. However we already use such approach in many places in the library (e.g. in [Scatter](https://github.com/f5/unovis/blob/main/packages/ts/src/components/scatter/types.ts)).

Reply:
@rokotyan That was my original thought actually, but I don't think it will work for primitive types (i.e. number), so I think wrapping it is necessary. How do we bypass this in Scatter? Can we give data as an array of numbers?

Reply:
@reb-dev Very good point!

It won't be possible to give data as an array of numbers to Scatter. Let's do the right thing for Donut and wrap it like you proposed in the beginning. Then there's no need to have the `_` prefix, right?
 
```ts
export type DonutDatum<Datum> = {
  datum: Datum;
  index: number;
}
```